src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -176,17 +176,26 @@ public FireStoreDirectoryEntry retrieveByPath(
         .orElse(null);
   }
 
-  public List<UUID> validateRefIds(TableServiceClient tableServiceClient, List<UUID> refIdArray) {
+  public List<String> validateRefIds(
+      TableServiceClient tableServiceClient, List<String> refIdArray) {
     logger.info("validateRefIds for {} file ids", refIdArray.size());
-    List<UUID> missingIds = new ArrayList<>();
-    for (UUID s : refIdArray) {
-      ListEntitiesOptions options =
-          new ListEntitiesOptions().setFilter(String.format("fileId eq '%s'", s.toString()));
-      if (!TableServiceClientUtils.tableHasEntries(tableServiceClient, TABLE_NAME, options)) {
-        missingIds.add(s);
-      }
-    }
-    return missingIds;
+    TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
+    return ListUtils.partition(refIdArray, MAX_FILTER_CLAUSES).stream()
+        .flatMap(
+            refIds -> {
+              String filter =
+                  refIds.stream()
+                      .map(refId -> String.format("fileId eq '%s'", refId))
+                      .collect(Collectors.joining(" or "));
+              ListEntitiesOptions options = new ListEntitiesOptions().setFilter(filter);
+              PagedIterable<TableEntity> entities = tableClient.listEntities(options, null, null);
```

Comment:
Is there a chance that there are no entities to be listed? If it's possible, I recommend doing [this "tableHasEntities](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/filedata/azure/tables/TableServiceClientUtils.java#L26)" check first. The TableClient REALLY loves NPEs.

Reply:
There definitely is that chance.

Reply:
Added a case to test ingesting a non-existent fileId before any files had been uploaded, and low and behold, an NPE appeared! I've added the check for entities to satisfy the test case.